pkg/blocks/basicstep.go

```diff
@@ -31,108 +31,35 @@ import (
 	"github.com/facebookincubator/ttpforge/pkg/logging"
 	"github.com/facebookincubator/ttpforge/pkg/outputs"
 	"go.uber.org/zap"
-	"gopkg.in/yaml.v3"
+)
+
+const (
+	ExecutorPython     = "python3"
+	ExecutorBash       = "bash"
+	ExecutorSh         = "sh"
+	ExecutorPowershell = "powershell"
+	ExecutorRuby       = "ruby"
+	ExecutorBinary     = "binary"
+	ExecutorCmd        = "cmd.exe"
 )
 
 // BasicStep is a type that represents a basic execution step.
 type BasicStep struct {
-	*Act        `yaml:",inline"`
-	Executor    string     `yaml:"executor,omitempty"`
-	Inline      string     `yaml:"inline,flow"`
-	Args        []string   `yaml:"args,omitempty,flow"`
-	CleanupStep CleanupAct `yaml:"cleanup,omitempty"`
+	actionDefaults `yaml:"-"`
+	Executor       string                  `yaml:"executor,omitempty"`
+	Inline         string                  `yaml:"inline,flow"`
+	Environment    map[string]string       `yaml:"env,omitempty"`
+	Outputs        map[string]outputs.Spec `yaml:"outputs,omitempty"`
 }
 
 // NewBasicStep creates a new BasicStep instance with an initialized Act struct.
 func NewBasicStep() *BasicStep {
-	return &BasicStep{
-		Act: &Act{
-			Type: StepBasic,
-		},
-	}
-}
-
-// UnmarshalYAML custom unmarshaler for BasicStep to handle decoding from YAML.
-func (b *BasicStep) UnmarshalYAML(node *yaml.Node) error {
```

Comment:
A core payoff of this changset is that we no longer need super verbose and repetitive custom UnmarshalYAML implementations for every single step

Reply:
This is a great win